core/amber/src/main/scala/edu/uci/ics/amber/engine/common/storage/VirtualDocument.scala

```diff
@@ -32,32 +32,33 @@ abstract class VirtualDocument[T >: Null <: AnyRef] {
   def get(): Iterator[T] = throw new NotImplementedError("get method is not implemented")
 
   /**
-    * append one data item to the document
+    * set ith item
     * @param item the data item
     */
-  def setItem(item: T): Unit =
+  def setItem(i: Int, item: T): Unit =
     throw new NotImplementedError("setItem method is not implemented")
 
   /**
-    * set ith item
+    * append one data item to the document
+    *
     * @param item the data item
     */
-  def setItem(i: Int, item: T): Unit =
+  def append(item: T): Unit =
     throw new NotImplementedError("setItem method is not implemented")
 
   /**
     * append data items from the iterator to the document
     * @param items iterator for the data item
     */
-  def write(items: Iterator[T]): Unit =
-    throw new NotImplementedError("write method is not implemented")
+  def append(items: Iterator[T]): Unit =
+    throw new NotImplementedError("append method is not implemented")
 
   /**
     * append the file content with an opened input stream
     * @param inputStream the data source input stream
     */
-  def write(inputStream: InputStream): Unit =
-    throw new NotImplementedError("write method is not implemented")
+  def append(inputStream: InputStream): Unit =
```

Comment:
it is better to call this function `appendStream` instead of `append` because we don't want the user to call this `append` together with other `append` variants.